### DISCUSSION TITLE
`amount` in HRA doesn't seem to work

### DISCUSSION THREAD

**Discussion by erplsf** [actions/actions-runner-controller#1406] [Questions] [answered]
Greetings, I have configuration for HRA as follows:

<img width="1249" alt="Screenshot 2022-04-29 at 15 22 59" src="https://user-images.githubusercontent.com/4072364/165953009-1bea171e-9b00-4984-85c9-92613b0c21ad.png">

My intention is to spin up 3 StatefulSet runners on each webhook delivery `workflow_job`. From manual testing and redelivering webhooks manually, I can see that only 1 pod is spinned up. Am I misunderstanding documentation [here](https://github.com/actions-runner-controller/actions-runner-controller#webhook-driven-scaling)?
> With the above example, the webhook server scales example-runners by 1 replica for 5 minutes on each check_run event with the type of created and the status of queued received.

Runner Controller version is `v0.22.3`.

**Comment by erplsf** [comment]
What also confuses me is that webhook server outputs logs like these (where I would expect amount to be 3):
`github-webhook-server 2022-04-29T13:30:59Z    INFO    controllers.webhookbasedautoscaler    scaled default-runner-autoscaler by 1`

**Comment by mumoshu** [comment, marked as answer]
@erplsf Hey! I don't remember the details already, but reading through our implementation Amount seems no-op. But hey, as long as you use `workflow_job` for scaling, you should have no need to customize it. I suppose you thought it would be a good idea to add 3 replicas on each workflow_job event because you use build matrix, right? You shouldn't need to do so, because as far as I remember, github produced workflow_job event for each job triggered by the matrix.

**Comment by erplsf** [comment]
Hi. Thanks for the info.

I increased the amount because our users complained about waiting for hours
in queue, and observed in real time that when three jobs start in the
matrix, only one pod was brought up.

**Reply by mumoshu** [reply]
@erplsf Hey. Could you try to browse github webhook evenst history so that (as far as I remember) you can see that each job in a matrix triggers a dedicated event?

I'm now suspecting that your issue might be related to https://github.com/actions-runner-controller/actions-runner-controller/issues/1321

**Reply by erplsf** [reply]
I'll try, thanks for the hint.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
